graylog2-web-interface/src/views/components/visualizations/ChartData.js

```diff
@@ -37,14 +37,19 @@ export const formatSeries = ({ valuesBySeries, xLabels }: {valuesBySeries: Objec
   ]);
 };
 
+export const getLeafsFromRows = (rows: Rows): Array<Leaf> => {
+  // $FlowFixMe: Somehow flow is unable to infer that the result consists only of Leafs.
+  return rows.filter(row => (row.source === 'leaf'));
+};
+
+export const getXLabelsFromLeafs = (leafs: Array<Leaf>): Array<any> => leafs.map(({ key }) => key);
```

Comment:
The return type of this should actually be `Array<Array<Key>>` instead of `Array<any>`. Although `Key` is currently `*`, we might be able/want to narrow it down in the future. Another alternative is to specify the type of `map` and do something like this:

```suggestion
export const getXLabelsFromLeafs = (leafs: Array<Leaf>) => leafs.map<Array<Key>>(({ key }) => key);
```